tests/testthat/test-export.R

```diff
@@ -0,0 +1,7 @@
+testthat::test_that("Listing print correctly", {
+  lsting <- as_listing(anl, key_cols = c("USUBJID"))
+  main_title(lsting) <- "this is some title"
+  main_footer(lsting) <- "this is some footer"
+  result <- export_as_txt(lsting, file = NULL)
+  testthat::expect_snapshot(cat(result))
```

Comment:
when using `export_as_txt(lsting, file=  NULL, lpp = 108, cpp = 125)`, got error 
```
non-null colwidths argument must have length ncol(lst) [48], got length 6
```

Reply:
Hi @clarkliming , can you give me an exmaple here?

when i try 

```
dat <- ex_adae
lsting <- as_listing(dat[1:25, ], key_cols = c("USUBJID", "AESOC")) %>%
    add_listing_col("AETOXGR") %>%
    add_listing_col("BMRKR1", format = "xx.x") %>%
    add_listing_col("AESER / AREL", fun = function(df) paste(df$AESER, df$AREL, sep = " / "))
main_title(lsting) <- "this is some title"
main_footer(lsting) <- "this is some footer"
export_as_txt(lsting, file=  NULL, lpp = 108, cpp = 125)
```

it seems working. thanks

Reply:
```
tmp_data <- ex_adae %>%
    dplyr::slice(1:30) %>%
    dplyr::distinct(USUBJID, AGE, BMRKR1, .keep_all = TRUE)

  lsting <- as_listing(tmp_data,
    key_cols = c("USUBJID", "AGE"),
    disp_cols = character()
  ) %>%
    add_listing_col("BMRKR1", format = "xx.x")
export_as_txt(lsting, file=  NULL, lpp = 108, cpp = 125)
```

got error 
```Error in export_as_txt(lsting, file = NULL, lpp = 108, cpp = 125) : 
  non-null colwidths argument must have length ncol(lst) [48], got length 3
```

package versions

[1] rlistings_0.1.1.9018 testthat_3.1.6       tibble_3.1.8         formatters_0.3.4.21

Reply:
cool! Thanks @clarkliming , it should be working now

```
[1] "—————————————————————————————————————————————————————————————\nUnique Subject Identifier   Age   Continous Level Biomarker 1\n—————————————————————————————————————————————————————————————\nAB12345-BRA-1-id-134        47                6.5            \nAB12345-BRA-1-id-141        35                7.5            \nAB12345-BRA-1-id-236        32                7.7            \nAB12345-BRA-1-id-265        25               10.3            \nAB12345-BRA-1-id-42         36                2.3            \nAB12345-BRA-1-id-65         25                7.3            \n\n—————————————————————————————————————————————————————————————\n"
```